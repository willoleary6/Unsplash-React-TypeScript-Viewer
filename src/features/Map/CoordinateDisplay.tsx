import React from "react";
import "./MapWidget.css";
import { coordinates } from "../../slices/GeoMap/GeoMapSlice";

interface coordinateProps {
    coordinates: coordinates;
    displayClassName: string;
}

export const CoordinateDisplay = ({
    coordinates,
    displayClassName: displayClassName,
}: coordinateProps): JSX.Element => {
    return (
        <>
            <div className={displayClassName}>
                <p className={"longitude-latitude-titlebox"}>Lat: {coordinates.lat}</p>
                <p className={"longitude-latitude-titlebox"}>Lng: {coordinates.lng}</p>
            </div>
        </>
    );
};
