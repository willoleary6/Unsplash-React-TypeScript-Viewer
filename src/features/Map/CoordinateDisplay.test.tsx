import React from "react";
import { CoordinateDisplay } from "./CoordinateDisplay";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
Enzyme.configure({ adapter: new Adapter() });

describe("CoordinateDisplay render testing", () => {
    it("renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        const wrapper = shallow(
            <CoordinateDisplay
                coordinates={{ lat: 0, lng: 0 }}
                displayClassName="latitude-value-container"
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
