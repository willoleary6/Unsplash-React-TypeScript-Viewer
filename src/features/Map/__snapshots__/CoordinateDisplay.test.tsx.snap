// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoordinateDisplay render testing renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<Fragment>
  <div
    className="latitude-value-container"
  >
    <p
      className="longitude-latitude-titlebox"
    >
      Lat: 
      0
    </p>
    <p
      className="longitude-latitude-titlebox"
    >
      Lng: 
      0
    </p>
  </div>
</Fragment>
`;
